doc/bn.tex

```diff
@@ -820,28 +820,32 @@ \subsection{Single Digit}
 \}
 \end{alltt} \end{small}
 
-\subsection{Long Constants}
+\subsection{Int32 and Int64 Constants}
 
-To set a constant that is the size of an ISO C ``unsigned long'' and larger than a single digit the following function
-can be used.
+To set a constant with 32 or 64 bits the following function can be used.
 
-\index{mp\_set\_int}
+\index{mp\_set\_sint}
 \begin{alltt}
-int mp_set_int (mp_int * a, unsigned long b);
+void mp_set_sint (mp_int * a, int32_t b);
+void mp_set_uint (mp_int * a, uint32_t b);
+void mp_set_sint64 (mp_int * a, int64_t b);
+void mp_set_uint64 (mp_int * a, uint64_t b);
 \end{alltt}
 
-This will assign the value of the 32-bit variable $b$ to the mp\_int $a$.  Unlike mp\_set() this function will always
-accept a 32-bit input regardless of the size of a single digit.  However, since the value may span several digits
-this function can fail if it runs out of heap memory.
-
-To get the ``unsigned long'' copy of an mp\_int the following function can be used.
+This will assign the value of the variable $b$ to the mp\_int $a$. The unsigned variants set a positive value, the signed variants set the sign. The value can be obtained again by calling the following function.
 
 \index{mp\_get\_int}
 \begin{alltt}
-unsigned long mp_get_int (mp_int * a);
+int32_t mp_get_sint (mp_int * a);
+uint32_t mp_get_uint (mp_int * a);
+uint32_t mp_get_mag (mp_int * a);
+int64_t mp_get_sint64 (mp_int * a);
+uint64_t mp_get_uint64 (mp_int * a);
+uint64_t mp_get_mag64 (mp_int * a);
 \end{alltt}
 
-This will return the 32 least significant bits of the mp\_int $a$.
+This will return either 32 or 64 least significant bits of the mp\_int $a$. The unsigned functions
+return negative values in two complement representation. If you only want to access the magnitude, use the mp\_get\_mag functions.
```

Comment:
"in a two complement representation" or "in a twos complement representation". I would prefer the latter as does [Google](https://books.google.com/ngrams/graph?content=twos+complement%2Ctwos-complement&case_insensitive=on&year_start=1800&year_end=2000&corpus=15&smoothing=3&share=&direct_url=t4%3B%2Ctwos%20complement%3B%2Cc0%3B%2Cs0%3B%3Btwos%20complement%3B%2Cc0%3B%3BTwos%20complement%3B%2Cc0%3B%3BTwos%20Complement%3B%2Cc0%3B%3BTWOS%20COMPLEMENT%3B%2Cc0%3B.t4%3B%2Ctwos%20-%20complement%3B%2Cc0%3B%2Cs0%3B%3Btwos%20-%20complement%3B%2Cc0%3B%3BTwos%20-%20complement%3B%2Cc0%3B%3BTwos%20-%20Complement%3B%2Cc0).

The *mag* functions return the magnitude which is the absolute value. Don't forget that documentation get's searched and glimpsed over quickly  and hence should contain more keywords than actually necessary.
E.g.:
"The \texttt{mp\_get\_mag} functions return the magnitude, the absolute value of the input."

Reply:
fixed